tests/Microsoft.Identity.Test.Unit.net45/Integration/DeviceCodeFlowIntegrationTest.cs

```diff
@@ -0,0 +1,152 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Test.Common;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.Support.UI;
+
+namespace Microsoft.Identity.Test.Unit.Integration
+{
+    // Important: do not install a NuGet package with the Chrome driver as it is a security risk.
+    // Instead, install the Chrome driver on the test machine
+
+    // Note: these tests require permission to a KeyVault Microsoft account;
+    // Please ignore them if you are not a Microsoft FTE, they will run as part of the CI build
+    [TestClass]
+    [TestCategory(TestCategories.Selenium)]
+    [TestCategory(TestCategories.LabAccess)]
+    public class DeviceCodeFlow
+    {
+        private const string UsernameHtmlId = "i0116";
```

Comment:
These values are available in the coreUITestConstants.

Reply:
Taking a dependency on that is complicated, I don't want to do it for a single test, so I'll leave this as is for now and add a TODO. 

If we decide to do more Selenium tests, I'll consolidate with the Xamarin UI tests.

Reply:
Should be possible to make at least a file reference even if you don't want to take a project dependency for now.